### PR TITLE
Hook on-disk vortex files into benchmarking

### PR DESCRIPTION
**Pull request by AdamGS**
Adding the Vortex on-file format to our benchmarking suite

### CODE REVIEW

File: vortex-datafusion/src/datatype.rs [hunk 1]
```diff
@@ -25,7 +25,7 @@ use vortex_dtype::{DType, Nullability, PType};
 ///
 /// This function will panic if the provided `dtype` is not a StructDType, or if the struct DType
 /// has top-level nullability.
-pub(crate) fn infer_schema(dtype: &DType) -> Schema {
```
Comment (robert3005) [inline vortex-datafusion/src/datatype.rs]: this can stay `pub(crate)`

File: vortex-serde/src/layouts/reader/builder.rs [hunk 2]
```diff
@@ -99,7 +99,7 @@ impl<R: VortexReadAt> VortexLayoutReaderBuilder<R> {
         len as usize
     }
 
-    async fn read_footer(&mut self) -> VortexResult<Footer> {
+    pub async fn read_footer(&mut self) -> VortexResult<Footer> {
```
Comment (robert3005) [inline vortex-serde/src/layouts/reader/builder.rs]: let's not make it pub for now

**Review by robert3005 — APPROVED** [review]
Let's revert the visibility modifications

File: vortex-datafusion/src/persistent/execution.rs [hunk 3]
```diff
@@ -28,19 +27,17 @@ impl VortexExec {
         projection: Option<&Vec<usize>>,
         predicate: Option<Arc<dyn PhysicalExpr>>,
     ) -> DFResult<Self> {
-        let partitioning = Partitioning::UnknownPartitioning(1);
+        let projected_schema = project_schema(&file_scan_config.file_schema, projection)?;
         let plan_properties = PlanProperties::new(
-            EquivalenceProperties::new(file_scan_config.file_schema.clone()),
-            partitioning,
+            EquivalenceProperties::new(projected_schema),
+            Partitioning::UnknownPartitioning(1),
```
Comment (robert3005) [inline vortex-datafusion/src/persistent/execution.rs:33]: we should figure this one out. Requires some extra config in the reader

Comment (AdamGS) [reply vortex-datafusion/src/persistent/execution.rs:33]: Right now, we do only use a single partition in the file scan as all files are in the same group.

File: vortex-serde/src/layouts/reader/builder.rs (hunk 2 above)
Comment (AdamGS) [reply vortex-serde/src/layouts/reader/builder.rs]: I need this for the benchmarks, I think in general its useful as a user to be able to read the file's schema.

File: vortex-datafusion/src/datatype.rs (hunk 1 above)
Comment (AdamGS) [reply vortex-datafusion/src/datatype.rs]: same as below basically - both needed for benchmarks and a generally useful API.

**Pull request merged**